tests/haskell_module/repl/haskell_module_repl_test.go

```diff
@@ -0,0 +1,171 @@
+package haskell_module_repl_test
```

Comment:
> Perhaps this should get automated to read from the fs somehow

We are going there slowly :). The current workspace representation inherits the one used by `rules_go`. The `rules_bazel_integration_test` do this, but has other issues which is incompatible with `rules_haskell` needs.  Anyway it worth to either adopt `rules_bazel_integration_test`, either implement the same approach on top of `rules_go` rules we are using now

> however writing test cases inline in a go file feels a bit weird

Reply:
I don't mind go (but of course Haskell would be more convenient :P). I was thinking more along the lines of having the file structure of the test on the file system, and then instead of having a inline string, reading them from the fs. However, this will
* definitely slow down tests
* make file management more complex?

On the other hand,
* it makes the structure easier to observe and inspect in my opinion - people are used to browsing a file system, not so much a custom (albeit extremely lightweight) format
* error reports are better - e.g. when bazel fails to parse a test file now we get a line in that file, but that doesn't correspond to the source line in the .go file.

Anyways, I'm fine with following the status quo for now - it's definitely workable. I was just raising these things to potentially start a discussion. I think a definite improvement would be somehow exposing a common `WORKSPACE` for all these tests, or at least the ones that don't do anything fancy with the `WORKSPACE`.